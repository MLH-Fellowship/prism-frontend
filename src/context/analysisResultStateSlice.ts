// TODO remove
/* eslint-disable no-console */
import { createAsyncThunk, createSlice, PayloadAction } from '@reduxjs/toolkit';
import { FeatureCollection, Position } from 'geojson';
import { get, find } from 'lodash';
import { CreateAsyncThunkTypes, RootState } from './store';
import {
  AggregationOperations,
  AsyncReturnType,
  BoundaryLayerProps,
  NSOLayerProps,
  ThresholdDefinition,
  WMSLayerProps,
  LegendDefinition,
} from '../config/types';
import {
  ApiData,
  BaselineLayerData,
  checkBaselineDataLayer,
  fetchApiData,
  generateFeaturesFromApiData,
} from '../utils/analysis-utils';
import { getWCSLayerUrl } from './layers/wms';
import { getBoundaryLayerSingleton, LayerDefinitions } from '../config/utils';
import { Extent } from '../components/MapView/Layers/raster-utils';
import { layerDataSelector } from './mapStateSlice';
import { LayerData, LayerDataParams, loadLayerData } from './layers/layer-data';
import { DataRecord, NSOLayerData } from './layers/nso';
import { BoundaryLayerData } from './layers/boundary';

type AnalysisResultState = {
  result?: AnalysisResult;
  error?: string;
  isLoading: boolean;
  isMapLayerActive: boolean;
};
export class AnalysisResult {
  key: number = Date.now();
  featureCollection: FeatureCollection;
  tableData: TableRow[];
  rawApiData?: object[];
  statistic: AggregationOperations;
  legend: LegendDefinition;
  hazardLayerId: WMSLayerProps['id'];
  baselineLayerId: NSOLayerProps['id'];

  constructor(
    tableData: TableRow[],
    featureCollection: FeatureCollection,
    hazardLayer: WMSLayerProps,
    baselineLayer: NSOLayerProps,
    statistic: AggregationOperations,
    rawApiData?: object[],
  ) {
    this.featureCollection = featureCollection;
    this.tableData = tableData;
    this.statistic = statistic;
    this.legend = baselineLayer.legend;
    this.rawApiData = rawApiData;

    this.hazardLayerId = hazardLayer.id;
    this.baselineLayerId = baselineLayer.id;
  }
  getHazardLayer(): WMSLayerProps {
    return LayerDefinitions[this.hazardLayerId] as WMSLayerProps;
  }
  getBaselineLayer(): NSOLayerProps {
    return LayerDefinitions[this.baselineLayerId] as NSOLayerProps;
  }
}

export type TableRow = {
  localName: string;
  name: string;
  baselineValue: DataRecord['value'];
  coordinates?: Position;
} & { [k in AggregationOperations]: number };

const initialState: AnalysisResultState = {
  isLoading: false,
  isMapLayerActive: true, // TODO toggle via UI
};

function generateTableFromApiData(
  aggregateData: AsyncReturnType<typeof fetchApiData>, // data from api
  // admin layer, both props and data. Aimed to closely match LayerData<BoundaryLayerProps>
  {
    layer: adminLayer,
    data: adminLayerData,
  }: { layer: BoundaryLayerProps; data: BoundaryLayerData },
  // baseline layer, both props and data
  {
    layer: { adminLevel },
    data: { layerData },
  }: { layer: NSOLayerProps; data: NSOLayerData },
): TableRow[] {
  // find the key that will let us reference the names of the bounding boxes.
  const adminLevelName =
    adminLayer.adminLevelNames[adminLevel - 1] || adminLayer.adminLevelNames[0];
  // for local name too.
  const adminLevelLocalName =
    adminLayer.adminLevelLocalNames[adminLevel - 1] ||
    adminLayer.adminLevelLocalNames[0];

  return aggregateData.map((row: any) => {
    // find feature from admin boundaries that closely matches this api row.
    // once we find it we can get the corresponding local name.
    const featureBoundary = find(
      adminLayerData.features,
      (feature: any) =>
        feature.properties[adminLevelName] === row[adminLevelName],
    );

    const name: string =
      featureBoundary?.properties?.[adminLevelName] || 'No Name';
    const localName: string =
      featureBoundary?.properties?.[adminLevelLocalName] || 'No Name';

    const baselineValue =
      layerData.find(({ adminKey }) => {
        return (featureBoundary?.properties?.[
          adminLayer.adminCode
        ] as string).startsWith(adminKey);
      })?.value || 'No Data';

    const tableRow: TableRow = {
      name,
      localName,
      mean: get(row, `stats_${AggregationOperations.Mean}`, 0),
      median: get(row, `stats_${AggregationOperations.Median}`, 0),
      baselineValue,
      coordinates: (featureBoundary?.geometry as any).coordinates[0][0][0], // TODO likely will not keep
    };
    return tableRow;
  });
}

export type AnalysisDispatchParams = {
  baselineLayer: NSOLayerProps;
  hazardLayer: WMSLayerProps;
  extent: Extent;
  threshold: ThresholdDefinition;
  date: ReturnType<Date['getTime']>; // just a hint to developers that we give a date number here, not just any number
  statistic: AggregationOperations; // we might have to deviate from this if analysis accepts more than what this enum provides
};
const apiUrl = 'https://prism-api.ovio.org/stats'; // TODO both needs to be stored somewhere
const adminJson =
  'https://prism-admin-boundaries.s3.us-east-2.amazonaws.com/mng_admin_boundaries.json';

export const requestAndStoreAnalysis = createAsyncThunk<
  AnalysisResult,
  AnalysisDispatchParams,
  CreateAsyncThunkTypes
>('analysisResultState/requestAndStoreAnalysis', async (params, api) => {
  const {
    hazardLayer,
    date,
    baselineLayer,
    extent,
    statistic,
    threshold,
  } = params;
  const baselineData = layerDataSelector(baselineLayer.id)(
    api.getState(),
  ) as LayerData<NSOLayerProps>;
  const adminBoundaries = getBoundaryLayerSingleton();
  const adminBoundariesData = layerDataSelector(adminBoundaries.id)(
    api.getState(),
  ) as LayerData<BoundaryLayerProps>;

  if (!adminBoundariesData) {
    throw new Error('Boundary Layer not loaded!');
  }
  // we force group by to be defined with &
  // eslint-disable-next-line camelcase
  const apiRequest: ApiData & { group_by: string } = {
    geotiff_url: getWCSLayerUrl({
      layer: hazardLayer,
      date,
      extent,
    }),
    zones_url:
      process.env.NODE_ENV === 'production'
        ? window.location.origin +
          getBoundaryLayerSingleton().path.replace('.', '')
        : adminJson,
    group_by:
      adminBoundaries.adminLevelNames[baselineLayer.adminLevel - 1] ||
      adminBoundaries.adminLevelNames[0],
  };
  const aggregateData = await fetchApiData(apiUrl, apiRequest);
  let baselineLayerData: BaselineLayerData;
  // if the baselineData doesn't exist, lets load it, otherwise check then load existing data.
  if (!baselineData) {
    const {
      payload: { data },
    } = (await api.dispatch(
      loadLayerData({ layer: baselineLayer, extent } as LayerDataParams<
        NSOLayerProps
      >),
    )) as { payload: { data: unknown } };

    // eslint-disable-next-line fp/no-mutation
    baselineLayerData = checkBaselineDataLayer(baselineLayer.id, data);
  } else {
    // eslint-disable-next-line fp/no-mutation
    baselineLayerData = checkBaselineDataLayer(
      baselineLayer.id,
      baselineData.data,
    );
  }

  const features = generateFeaturesFromApiData(
    aggregateData,
    hazardLayer,
    baselineLayerData,
    apiRequest.group_by,
    statistic,
    threshold,
  );

  const tableRows: TableRow[] = generateTableFromApiData(
    aggregateData,
    adminBoundariesData,
    { layer: baselineLayer, data: baselineLayerData },
  );

  return new AnalysisResult(
    tableRows,
    {
      ...adminBoundariesData.data,
      features,
    },
    hazardLayer,
    baselineLayer,
    statistic,
    aggregateData, // TODO we only pass this to make debugging easier - not used anywhere
  );
});

export const analysisResultSlice = createSlice({
  name: 'analysisResultState',
  initialState,
  reducers: {
    setIsMapLayerActive: (state, { payload }: PayloadAction<boolean>) => ({
      ...state,
      isMapLayerActive: payload,
    }),
  },
  extraReducers: builder => {
    builder.addCase(
      requestAndStoreAnalysis.fulfilled,
      (
        { result, ...rest },
        { payload }: PayloadAction<AnalysisResult>,
      ): AnalysisResultState => ({
        ...rest,
        isLoading: false,
        result: payload,
      }),
    );

    builder.addCase(
      requestAndStoreAnalysis.rejected,
      (state, action): AnalysisResultState => ({
        ...state,
        isLoading: false, // TODO
        error: action.error.message
          ? action.error.message
          : action.error.toString(),
      }),
    );

    builder.addCase(
      requestAndStoreAnalysis.pending,
      (state): AnalysisResultState => ({
        ...state, // TODO
        isLoading: true,
      }),
    );
  },
});

// Getters
export const analysisResultSelector = (
  state: RootState,
): AnalysisResult | undefined => state.analysisResultState.result;

export const isAnalysisLoadingSelector = (state: RootState): boolean =>
  state.analysisResultState.isLoading;

export const isAnalysisLayerActiveSelector = (state: RootState): boolean =>
  state.analysisResultState.isMapLayerActive;

// Setters
export const { setIsMapLayerActive } = analysisResultSlice.actions;

export default analysisResultSlice.reducer;
