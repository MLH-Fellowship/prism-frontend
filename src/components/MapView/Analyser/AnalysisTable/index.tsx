import React, { useState } from 'react';
import { invert } from 'lodash';
import {
  Button,
  createStyles,
  Table,
  TableBody,
  TableCell,
  TableContainer,
  TableHead,
  TablePagination,
  TableRow,
  Typography,
  withStyles,
  WithStyles,
} from '@material-ui/core';
import { useDispatch } from 'react-redux';
import {
  AnalysisResult,
  TableRow as AnalysisTableRow,
} from '../../../../context/analysisResultStateSlice';
import { showPopup } from '../../../../context/tooltipStateSlice';
import { AggregationOperations } from '../../../../config/types';
import { downloadCSVFromTableData } from '../../../../utils/analysis-utils';

export type Column = {
  id: keyof AnalysisTableRow;
  label: string;
  format?: (value: number) => string;
};

function AnalysisTable({ classes, analysisResult }: AnalysisTableProps) {
  const [page, setPage] = useState(0);
  const [rowsPerPage, setRowsPerPage] = useState(10);

  const { tableData, statistic } = analysisResult;
  const baselineLayerTitle = analysisResult.getBaselineLayer().title;

  const dispatch = useDispatch();

  const handleChangePage = (event: unknown, newPage: number) => {
    setPage(newPage);
  };

  const handleChangeRowsPerPage = (
    event: React.ChangeEvent<HTMLInputElement>,
  ) => {
    setRowsPerPage(+event.target.value);
    setPage(0);
  };

  const columns: Column[] = [
    {
      id: 'localName',
      label: 'Local Name',
    },
    {
      id: 'name',
      label: 'Name',
    },
    {
      id: statistic,
      label: invert(AggregationOperations)[statistic], // invert maps from computer name to display name.
      format: (value: number) => value.toLocaleString('en-US'),
    },

    {
      id: 'baselineValue',
      label: baselineLayerTitle,
      format: (value: number | string) => value.toLocaleString('en-US'),
    },
  ];
  return (
    <div>
      <TableContainer className={classes.tableContainer}>
        <Table stickyHeader aria-label="sticky table">
          <TableHead>
            <TableRow>
              {columns.map(column => (
                <TableCell key={column.id} className={classes.tableHead}>
                  {column.label}
                </TableCell>
              ))}
            </TableRow>
          </TableHead>
          <TableBody>
            {tableData
              .slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
              .map(row => {
                return (
                  <TableRow
                    hover
                    role="checkbox"
                    tabIndex={-1}
                    key={row.name}
                    onClick={() => {
                      // TODO if we decide to keep, add popup data?
                      if (row.coordinates) {
                        dispatch(
                          showPopup({
                            coordinates: row.coordinates,
                            locationName: row.name,
                          }),
                        );
                      }
                    }}
                    style={{ cursor: row.coordinates ? 'pointer' : 'none' }}
                  >
                    {columns.map(column => {
                      const value = row[column.id];
                      return (
                        <TableCell key={column.id}>
                          {column.format && typeof value === 'number'
                            ? column.format(value)
                            : value}
                        </TableCell>
                      );
                    })}
                  </TableRow>
                );
              })}
          </TableBody>
        </Table>
      </TableContainer>
      <TablePagination
        rowsPerPageOptions={[10, 25, 100]}
        component="div"
        count={tableData.length}
        rowsPerPage={rowsPerPage}
        page={page}
        onChangePage={handleChangePage}
        onChangeRowsPerPage={handleChangeRowsPerPage}
      />
      <Button
        className={classes.innerAnalysisButton}
        onClick={() => downloadCSVFromTableData(analysisResult, columns)}
      >
        <Typography variant="body2">Download</Typography>
      </Button>
    </div>
  );
}

const styles = () =>
  createStyles({
    tableContainer: {
      border: '2px solid',
      width: '600px',
      maxWidth: '100vw',
    },
    tableHead: {
      backgroundColor: '#3d474a',
    },
    innerAnalysisButton: {
      backgroundColor: '#3d474a',
    },
  });

interface AnalysisTableProps extends WithStyles<typeof styles> {
  analysisResult: AnalysisResult;
}

export default withStyles(styles)(AnalysisTable);
