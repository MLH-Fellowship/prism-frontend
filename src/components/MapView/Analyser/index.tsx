/* eslint-disable no-console */
import React, { useMemo, useState } from 'react';
import {
  Button,
  createStyles,
  FormControl,
  FormControlLabel,
  Radio,
  RadioGroup,
  Theme,
  Typography,
  withStyles,
  WithStyles,
} from '@material-ui/core';
import { ArrowDropDown, Assessment } from '@material-ui/icons';
import { useDispatch, useSelector } from 'react-redux';
import { find, map } from 'lodash';
import bbox from '@turf/bbox';
import {
  getBoundaryLayerSingleton,
  LayerDefinitions,
} from '../../../config/utils';
import {
  AggregationOperations,
  BoundaryLayerProps,
  NSOLayerProps,
  WMSLayerProps,
} from '../../../config/types';
import { LayerData } from '../../../context/layers/layer-data';
import { layerDataSelector } from '../../../context/mapStateSlice';
import { Extent } from '../Layers/raster-utils';
import { availableDatesSelector } from '../../../context/serverStateSlice';
import {
  AnalysisDispatchParams,
  analysisResultSelector,
  isAnalysisLoadingSelector,
  requestAndStoreAnalysis,
} from '../../../context/analysisResultStateSlice';
import AnalysisTable from './AnalysisTable';

const layers = Object.values(LayerDefinitions);
const baselineLayers = layers.filter(
  (layer): layer is NSOLayerProps => layer.type === 'nso',
);
const hazardLayers = layers.filter(
  (layer): layer is WMSLayerProps => layer.type === 'wms',
);

const boundaryLayer = getBoundaryLayerSingleton();

function Analyser({ classes }: AnalyserProps) {
  const dispatch = useDispatch();
  const boundaryLayerData = useSelector(layerDataSelector(boundaryLayer.id)) as
    | LayerData<BoundaryLayerProps>
    | undefined;
  const availableDates = useSelector(availableDatesSelector);
  const analysisResult = useSelector(analysisResultSelector);
  const isAnalysisLoading = useSelector(isAnalysisLoadingSelector);

  const [openAnalyserForm, setOpenAnalyserForm] = useState(false);
  // const [analyserOption, setAnalyserOption] = useState('new');
  const [hazardLayerId, setHazardLayerId] = useState(
    hazardLayers[0].id as string,
  );
  const [statistic, setStatistic] = useState(AggregationOperations.mean);
  const [baselineLayerId, setBaselineLayerId] = useState(
    baselineLayers[0].id as string,
  );

  const [openResult, setOpenResult] = useState(false);

  const onOptionChange = (setterFunc: (val: any) => void) => (
    event: React.ChangeEvent<HTMLInputElement>,
  ) => {
    setterFunc((event.target as HTMLInputElement).value);
  };

  const adminBoundariesExtent = useMemo(() => {
    if (!boundaryLayerData)
      // not loaded yet. Should be loaded in MapView
      return undefined;
    return bbox(boundaryLayerData.data) as Extent; // we get extents of admin boundaries to give to the api.
  }, [boundaryLayerData]);

  const hazardLayerOptions = map(hazardLayers, hazardLayer => {
    return (
      <FormControlLabel
        key={hazardLayer.id}
        value={hazardLayer.id}
        control={<Radio className={classes.radioOptions} size="small" />}
        label={hazardLayer.title}
      />
    );
  });
  const baselineLayerOptions = map(baselineLayers, baselineLayer => {
    return (
      <FormControlLabel
        key={baselineLayer.id}
        value={baselineLayer.id}
        control={<Radio className={classes.radioOptions} size="small" />}
        label={baselineLayer.title}
      />
    );
  });
  const statisticOptions = map(Object.keys(AggregationOperations), stat => {
    return (
      <FormControlLabel
        key={stat}
        value={stat}
        control={<Radio className={classes.radioOptions} size="small" />}
        label={stat}
      />
    );
  });

  const runAnalyser = async () => {
    if (!adminBoundariesExtent) return; // hasn't been calculated yet

    const selectedHazardLayer = find(hazardLayers, {
      id: hazardLayerId,
    }) as WMSLayerProps;
    const selectedBaselineLayer = find(baselineLayers, {
      id: baselineLayerId,
    }) as NSOLayerProps;
    const params: AnalysisDispatchParams = {
      hazardLayer: selectedHazardLayer,
      baselineLayer: selectedBaselineLayer,
      date: availableDates[selectedHazardLayer.serverLayerName][0], // TODO load from ui
      statistic,
      extent: adminBoundariesExtent,
      threshold: {},
    };

    const data = await dispatch(requestAndStoreAnalysis(params));

    // TODO remove
    // eslint-disable-next-line no-console
    console.log(data);
  };

  return (
    <div className={classes.analyser}>
      <Button
        variant="contained"
        color="primary"
        onClick={() => {
          setOpenAnalyserForm(!openAnalyserForm);
          setOpenResult(false);
        }}
      >
        <Assessment style={{ marginRight: '10px' }} />
        <Typography variant="body2">Run Analysis</Typography>
        <ArrowDropDown />
      </Button>

      {isAnalysisLoading || analysisResult
        ? [
            <Button
              key="show"
              variant="contained"
              color="primary"
              className={classes.analyserButton}
              onClick={() => {
                setOpenAnalyserForm(false);
                setOpenResult(!openResult);
              }}
              disabled={isAnalysisLoading}
            >
              <Typography variant="body2">Show Result</Typography>
            </Button>,
            <Button
              key="download"
              variant="contained"
              color="primary"
              className={classes.analyserButton}
              disabled={isAnalysisLoading}
            >
              <Typography variant="body2">Download</Typography>
            </Button>,
          ]
        : ''}

      <div
        className={classes.analyserMenu}
        style={{
          // eslint-disable-next-line no-nested-ternary
          width: openAnalyserForm ? '40vw' : openResult ? 'min-content' : 0,
          padding: openAnalyserForm || openResult ? 10 : 0,
        }}
      >
        {openAnalyserForm ? (
          <div>
            {/* <FormControl component="fieldset">
          <RadioGroup value={analyserOption} onChange={onAnalyserOptionChange} row>
            <FormControlLabel
              value="new"
              control={
                <Radio className={classes.analyserOptions} size="small" />
              }
              label="Create a new analysis"
            />
            <FormControlLabel
              value="pre-configure"
              control={
                <Radio className={classes.analyserOptions} size="small" />
              }
              label="Run a pre-configured analysis"
            />
            <FormControlLabel
              value="generate"
              control={
                <Radio className={classes.analyserOptions} size="small" />
              }
              label="Generate spatial statistics"
            />
          </RadioGroup>
        </FormControl> */}
            <div className={classes.newAnalyserContainer}>
              <div>
                <Typography variant="body2">
                  Step 1 - Choose a hazard Layer:
                </Typography>

                <FormControl component="div">
                  <RadioGroup
                    name="hazardLayer"
                    value={hazardLayerId}
                    onChange={onOptionChange(setHazardLayerId)}
                  >
                    {hazardLayerOptions}
                  </RadioGroup>
                </FormControl>
              </div>
              <div>
                <Typography variant="body2">
                  Step 2 - Select a statistic:
                </Typography>

                <FormControl component="div">
                  <RadioGroup
                    name="statistics"
                    value={statistic}
                    onChange={onOptionChange(setStatistic)}
                    row
                  >
                    {statisticOptions}
                  </RadioGroup>
                </FormControl>
              </div>
              <div>
                <Typography variant="body2">
                  Step 3 - Choose a baseline Layer:
                </Typography>
                <FormControl component="div">
                  <RadioGroup
                    name="baselineLayer"
                    value={baselineLayerId}
                    onChange={onOptionChange(setBaselineLayerId)}
                  >
                    {baselineLayerOptions}
                  </RadioGroup>
                </FormControl>
              </div>
            </div>
            <Button
              className={classes.innerAnalysisButton}
              onClick={runAnalyser}
            >
              <Typography variant="body2">Run Analysis</Typography>
            </Button>
            {isAnalysisLoading || analysisResult ? (
              <>
                <Button
                  className={classes.innerAnalysisButton}
                  disabled={isAnalysisLoading}
                >
                  <Typography variant="body2">Show Result</Typography>
                </Button>

                <Button
                  className={classes.innerAnalysisButton}
                  disabled={isAnalysisLoading}
                >
                  <Typography variant="body2">Download</Typography>
                </Button>
              </>
            ) : (
              ''
            )}
          </div>
        ) : (
          <div>
            <FormControl component="div">
              <RadioGroup row>
                <FormControlLabel
                  value="mapview"
                  control={
                    <Radio className={classes.radioOptions} size="small" />
                  }
                  label="Map View"
                />
                <FormControlLabel
                  value="tableview"
                  control={
                    <Radio className={classes.radioOptions} size="small" />
                  }
                  label="Table View"
                />
              </RadioGroup>
            </FormControl>
            {analysisResult && analysisResult.tableData ? (
              <AnalysisTable tableData={analysisResult.tableData} />
            ) : (
              ''
            )}
          </div>
        )}
      </div>
    </div>
  );
}

const styles = (theme: Theme) =>
  createStyles({
    analyser: {
      zIndex: theme.zIndex.drawer,
      position: 'absolute',
      top: 2,
      left: 2,
      textAlign: 'left',
    },
    analyserMenu: {
      backgroundColor: '#5A686C',
      maxWidth: '100vw',
      color: 'white',
      overflowX: 'hidden',
      // transition: 'width 0.5s ease-in-out',
      whiteSpace: 'nowrap',
      borderTopRightRadius: '10px',
      borderBottomRightRadius: '10px',
      height: 'auto',
      maxHeight: '60vh',
    },
    analyserButton: {
      height: '36px',
      'margin-left': '3px',
    },
    analyserOptions: {
      color: 'white',
      padding: '2px 5px',
    },
    newAnalyserContainer: {
      padding: '5px',
      marginTop: '10px',
    },
    radioOptions: {
      color: 'white',
      padding: '2px 10px 2px 20px',
    },
    innerAnalysisButton: {
      backgroundColor: '#3d474a',
    },
  });

interface AnalyserProps extends WithStyles<typeof styles> {}

export default withStyles(styles)(Analyser);
