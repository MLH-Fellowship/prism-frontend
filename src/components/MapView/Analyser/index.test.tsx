import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';

import Analyser from '.';
import { store } from '../../../context/store';

jest.mock('../../../utils/LayerDropdown', () => 'mock-Layer-Dropdown');

test('renders as expected', () => {
  const { container } = render(
    <Provider store={store}>
      <Analyser />
    </Provider>,
  );
  expect(container).toMatchSnapshot();
});
