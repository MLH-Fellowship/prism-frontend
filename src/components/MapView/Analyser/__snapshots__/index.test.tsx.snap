// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as expected 1`] = `
<div>
  <div
    class="Analyser-analyser-1"
  >
    <mock-button
      color="primary"
      variant="contained"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        style="margin-right: 10px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7v-7h2v7zm4 0h-2V7h2v10zm4 0h-2v-4h2v4z"
        />
      </svg>
      <mock-typography
        variant="body2"
      >
        Run Analysis
      </mock-typography>
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M7 10l5 5 5-5z"
        />
      </svg>
    </mock-button>
    <div
      class="Analyser-analyserMenu-2"
      style="padding: 10px;"
    >
      <div>
        <div
          class="Analyser-newAnalyserContainer-5"
        >
          <div
            class="Analyser-analyserOptions-4"
          >
            <mock-typography
              variant="body2"
            >
              Hazard Layer
            </mock-typography>
            <mock-layer-dropdown
              classname="Analyser-selector-9"
              title="Hazard Layer"
              type="wms"
              value="pasture_anomaly"
            />
          </div>
          <div
            class="Analyser-analyserOptions-4"
          >
            <mock-typography
              variant="body2"
            >
              Statistic
            </mock-typography>
            <div
              class="MuiFormControl-root"
            >
              <div
                class="MuiFormGroup-root MuiFormGroup-row"
                role="radiogroup"
              >
                <label
                  class="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-38 MuiRadio-root MuiRadio-colorSecondary Analyser-radioOptions-6 PrivateSwitchBase-checked-39 Mui-checked MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        checked=""
                        class="PrivateSwitchBase-input-41"
                        name="statistics"
                        type="radio"
                        value="mean"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-54 PrivateRadioButtonIcon-checked-56"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55 MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <mock-typography
                    classname="MuiFormControlLabel-label"
                    component="span"
                  >
                    Mean
                  </mock-typography>
                </label>
                <label
                  class="MuiFormControlLabel-root"
                >
                  <span
                    aria-disabled="false"
                    class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-38 MuiRadio-root MuiRadio-colorSecondary Analyser-radioOptions-6 MuiIconButton-colorSecondary"
                  >
                    <span
                      class="MuiIconButton-label"
                    >
                      <input
                        class="PrivateSwitchBase-input-41"
                        name="statistics"
                        type="radio"
                        value="median"
                      />
                      <div
                        class="PrivateRadioButtonIcon-root-54"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
                          />
                        </svg>
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root PrivateRadioButtonIcon-layer-55 MuiSvgIcon-fontSizeSmall"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M8.465 8.465C9.37 7.56 10.62 7 12 7C14.76 7 17 9.24 17 12C17 13.38 16.44 14.63 15.535 15.535C14.63 16.44 13.38 17 12 17C9.24 17 7 14.76 7 12C7 10.62 7.56 9.37 8.465 8.465Z"
                          />
                        </svg>
                      </div>
                    </span>
                    <span
                      class="MuiTouchRipple-root"
                    />
                  </span>
                  <mock-typography
                    classname="MuiFormControlLabel-label"
                    component="span"
                  >
                    Median
                  </mock-typography>
                </label>
              </div>
            </div>
          </div>
          <div
            class="Analyser-analyserOptions-4"
          >
            <mock-typography
              variant="body2"
            >
              Baseline Layer
            </mock-typography>
            <mock-layer-dropdown
              classname="Analyser-selector-9"
              title="Baseline Layer"
              type="nso"
              value="population_below_poverty"
            />
          </div>
          <div
            class="Analyser-analyserOptions-4"
          >
            <mock-typography
              variant="body2"
            >
              Threshold
            </mock-typography>
            <mock-textfield
              classname="Analyser-numberField-10"
              error="false"
              id="filled-number"
              label="Min"
              type="number"
              value=""
              variant="filled"
            />
            <mock-textfield
              classname="Analyser-numberField-10"
              id="filled-number"
              label="Max"
              type="number"
              value=""
              variant="filled"
            />
          </div>
          <div
            class="Analyser-analyserOptions-4"
          >
            <mock-typography
              variant="body2"
            >
              Date
            </mock-typography>
            <div
              class="react-datepicker-wrapper"
            >
              <div
                class="react-datepicker__input-container"
              >
                <div
                  class="MuiInputBase-root MuiInput-root MuiInput-underline"
                >
                  <input
                    class="MuiInputBase-input MuiInput-input"
                    type="text"
                    value=""
                  />
                </div>
              </div>
            </div>
          </div>
        </div>
        <mock-button
          classname="Analyser-innerAnalysisButton-7"
          disabled="true"
        >
          <mock-typography
            variant="body2"
          >
            Run Analysis
          </mock-typography>
        </mock-button>
      </div>
    </div>
  </div>
</div>
`;
